### ISSUE TITLE
Checking activation of different background picture

### ISSUE THREAD

**Issue by elfring** [Xubuntu/lightdm-gtk-greeter#106]
I am trying the software “LightDM GTK+ Greeter version 2.0.8-1.fc34” out.
The file “/etc/lightdm/lightdm-gtk-greeter.conf” contains the following specifications for my system.

```
…
background = /usr/share/backgrounds/default.png
…
user-background = false
…
```

This selected picture is displayed for a very short moment for login attempts.
But I am observing that it is immediately replaced by a different picture (which I am using for my XFCE user session for a while).
:thinking: I wonder about this login image replacement when user backgrounds should not be applied at the moment.

**Comment by elfring** [comment]
Will [another report](https://github.com/canonical/lightdm/issues/205 "Disabling \"Draw user backgrounds\" does not come to effect") trigger further development and configuration considerations accordingly?

**Comment by bluesabre** [comment]
The `user-background` setting seems to be working for me. It's possible that your lightdm-gtk-greeter.conf file is not being read. Can you paste the output of:

```
sudo grep "\.conf" /var/log/lightdm/seat*-greeter.log
```

It should look something like this.
```
$ sudo grep "\.conf" /var/log/lightdm/seat*-greeter.log
** Message: 03:38:00.484: [Configuration] Reading file: /usr/share/lightdm/lightdm-gtk-greeter.conf.d/01_ubuntu.conf
** Message: 03:38:00.484: [Configuration] Reading file: /usr/share/lightdm/lightdm-gtk-greeter.conf.d/30_xubuntu.conf
** Message: 03:38:00.484: [Configuration] Reading file: /usr/local/etc/lightdm/lightdm-gtk-greeter.conf
** Message: 03:38:34.040: [Configuration] Reading file: /usr/share/lightdm/lightdm-gtk-greeter.conf.d/01_ubuntu.conf
** Message: 03:38:34.040: [Configuration] Reading file: /usr/share/lightdm/lightdm-gtk-greeter.conf.d/30_xubuntu.conf
** Message: 03:38:34.040: [Configuration] Reading file: /usr/local/etc/lightdm/lightdm-gtk-greeter.conf
** Message: 03:40:04.539: [Configuration] Reading file: /usr/share/lightdm/lightdm-gtk-greeter.conf.d/01_ubuntu.conf
** Message: 03:40:04.539: [Configuration] Reading file: /usr/share/lightdm/lightdm-gtk-greeter.conf.d/30_xubuntu.conf
** Message: 03:40:04.539: [Configuration] Reading file: /usr/local/etc/lightdm/lightdm-gtk-greeter.conf
** Message: 03:47:10.131: [Configuration] Reading file: /usr/share/lightdm/lightdm-gtk-greeter.conf.d/01_ubuntu.conf
** Message: 03:47:10.131: [Configuration] Reading file: /usr/share/lightdm/lightdm-gtk-greeter.conf.d/30_xubuntu.conf
** Message: 03:47:10.131: [Configuration] Reading file: /usr/local/etc/lightdm/lightdm-gtk-greeter.conf
```

**Comment by elfring** [comment]
> It's possible that your lightdm-gtk-greeter.conf file is not being read.

Why would such a questionable software behaviour happen (while I could look at available settings by some tools)? :thinking:

**Comment by bluesabre** [comment]
It could happen as a result of how software is packaged and installed. The greeter reads from multiple files. The tools that write the configuration only write to a single file. If they're not using the same files, changing settings won't work reliably if at all.

**Comment by elfring** [comment]
:thinking:
* Can such information be clarified any further?
* Would you like share more experiences about questionable behaviour in affected software areas?

**Comment by bluesabre** [comment]
It's intentional behavior, intended to support distributions. In my output above,

```
** Message: 03:47:10.131: [Configuration] Reading file: /usr/share/lightdm/lightdm-gtk-greeter.conf.d/01_ubuntu.conf
** Message: 03:47:10.131: [Configuration] Reading file: /usr/share/lightdm/lightdm-gtk-greeter.conf.d/30_xubuntu.conf
** Message: 03:47:10.131: [Configuration] Reading file: /usr/local/etc/lightdm/lightdm-gtk-greeter.conf
```

You can see that the greeter is reading each configuration file in order from `/usr/share/lightdm/lightdm-gtk-greeter.conf.d`. It's also reading from `/usr/local/etc/lightdm/lightdm-gtk-greeter.conf` because my local copy was installed to the `/usr/local/` prefix. If it was installed without a prefix, it would read from `/etc/lightdm/lightdm-gtk-greeter.conf` instead.

Because my greeter is installed to `/usr/local` but the settings manager is installed without a prefix, I have the scenario where the settings manager is writing to `/etc/lightdm/lightdm-gtk-greeter.conf` but the greeter isn't reading it. I created a symbolic link between `/usr/local/etc/lightdm/lightdm-gtk-greeter.conf` and `/etc/lightdm/lightdm-gtk-greeter.conf` to ensure they're using the same file.

To summarize, if the settings are not applying but you are using the versions of the software packaged by Fedora, you should consider reporting a bug on the [Fedora bug tracker](https://bugzilla.redhat.com/) instead.

**Comment by elfring** [comment]
>To summarize, if the settings are not applying but you are using the versions of the software packaged by …

Did you hear about any related open issues (and corresponding adjustments) for available software distributions?

**Comment by bluesabre** [comment]
No. Everything seems to be working here, at least on Xubuntu. Please paste the output of
```
sudo grep "\.conf" /var/log/lightdm/seat*-greeter.log
```
or I will not be able to assist you further.

**Comment by elfring** [comment]
:thought_balloon: I hope that software distributions will improve their packages (also for this issue).
